lib/components/kernel-monitor.js

```diff
@@ -66,22 +54,108 @@ const KernelMonitor = observer(({ store }: { store: store }) => {
       </ul>
     );
   }
-  const grouped = _.groupBy(
-    store.runningKernels,
-    kernel => kernel.gatewayName || "Local"
-  );
-  return (
-    <div className="kernel-monitor">
-      {_.map(grouped, (kernels, group) => (
-        <MonitorSection
-          store={store}
-          kernels={kernels}
-          group={group}
-          key={group}
-        />
-      ))}
-    </div>
-  );
+
+  const data = _.map(store.runningKernels, kernel => {
+    return {
+      gateway: kernel.gatewayName || "Local",
```

Comment:
This line needs fixing to support `WSKernel`. Maybe `kernel.transport.gatewayName` if flow allows it.

Reply:
Thanks for pointing this out ! I totally forget to look back to this part.

Flow doesn't allow `kernel.transport.gatewayName` and only `Kernel` & `WSKernel` has that property, which seems really weird.

 Now I find the _property_ `gatewayName` is only used in existing-kernel-picker.js and there is no need for `Kernel` object to have this property but only `WSKernel` need to have it.